### PR TITLE
[GPU] Add reorder between FC matmal and reshape

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Fix input layout of reshape primitive:
 - *Add reorder between FC matmal and reshape*
 - *Add 5d/6d format for only_gemm_users in case of int8 model*
### Tickets:
 - *84182*

### CODE REVIEW

**Comment by yeonbok** [comment]
Seems that we dont have corresponding functional tests  5D =>5D or 6D => 6D. Could you add them?

**Comment by kelvinchoi-intel** [comment]
> Added functional tests for matmul 5D and 6D

File: src/plugins/intel_gpu/src/plugin/ops/matmul.cpp [hunk 1]
```diff
@@ -157,9 +157,32 @@ static void CreateMatMulOp(Program& p, const std::shared_ptr<ngraph::op::v0::Mat
 
         auto lastLayerName = layerName;
         if (reshape_fc) {
-            auto outputShape = tensor_from_dims(op->get_output_shape(0));
             auto outReshapeName = layerName + "_cldnn_out_reshape";
-            auto outReshapePrim = cldnn::reshape(outReshapeName, layerName, outputShape, op->get_friendly_name());
+
+            // add reorder
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/matmul.cpp:162]: And how about applying this change only when outDims.size() > 4 ?

Comment (yeonbok) [reply src/plugins/intel_gpu/src/plugin/ops/matmul.cpp:162]: 1. To reduce the effort of such reduction as many as possible 
2. To show the intention of adding reorder more clearly here

**Review by yeonbok — APPROVED** [review]

**Pull request merged**